draft-irtf-nwcrg-coding-and-congestion-01.txt

```diff
@@ -1,528 +1,448 @@
+
+   The coded packets are sent within what the congestion window allows.
+
+   TODO : ADD MORE TEXT
+
+5.  FEC within the transport
+
+   Figure 4 illustrates the FEC within the transport case.
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+Kuhn, et al.              Expires July 22, 2020                 [Page 5]
+
+Internet-Draft            Coding and congestion             January 2020
+
+
+                          |
+                          | data, requirements
+                          |
+                          v
+           +---------------------+
+           | Transport           |
+           | +-----+  +-----+    | coded data
+           | | FEC |  | CC  |    |
+           | +-----+  +-----+    | ----->
+           |                     |
+           +---------------------+
+                   ^          ^
+                   |          |
+           information    network
+           about          measurements
+           repaired ratio
+
+                      Figure 4: FEC in the transport
+
+   The coded packets are sent within what the congestion window allows,
+   such as in [CTCP].  Examples of the solution would be sending coded
+   packets when there is no more data to transmit or preferably send
+   coded packets instead of the following packets in the send buffer.
+
+   TODO : ADD MORE TEXT
+
+6.  FEC below the transport
+
+   Figure 5 illustrates the FEC below the transport case.
+
+                  |                  --------                  ^
+                  | data             |      | data            | coded
+                  | requirements     |      |                 | data
+                  v                  |      v                 |
+   +-----------------------------+   |    +-------------------------+
+   | Transport (including CC)    |        |           FEC           |
+   +-----------------------------+   |    +-------------------------+
+     ^                   |           |              ^
+     | network           |           |              | information
+     | measurements      |           |              | about repair
+     |                   -------------              | ratio
+
+                     Figure 5: FEC below the transport
+
+   The coded packets are sent on top of what is allowed by the
+   congestion control.  Examples of the solution could be adding a given
+   percentage of the congestion window as supplementary packets or
+   sending a given amount of coded packets at a given rate.  The
+
+
+
+Kuhn, et al.              Expires July 22, 2020                 [Page 6]
+
+Internet-Draft            Coding and congestion             January 2020
+
+
+   redundancy flow can be decorrelated from the congestion control that
+   manages source packets: a secondary congestion control can be
+   introduced, such as in coupled congestion control for RTP media
+   [I-D.ietf-rmcat-coupled-cc].  An example would be to exploit a lower
+   than best-effort congestion control [RFC6297].
```

Comment:
I suggest to replace "a secondary congestion control can be introduced" with "a secondary congestion control could be introduced underneath the FEC layer".

Reply:
ACK